### PR TITLE
commit_walker: simplify implementation.

### PR DESCRIPTION
**Pull request by smola**
* No recursive calls.
* Check if seen before adding to the queue.

### CODE REVIEW

File: commit_walker.go [hunk 1]
```diff
@@ -6,62 +6,40 @@ import (
...
 
-type commitWalker struct {
-	seen  map[core.Hash]bool
-	stack []*CommitIter
-	start *Commit
-	cb    func(*Commit) error
-}
-
 // WalkCommitHistory walks the commit history
 func WalkCommitHistory(c *Commit, cb func(*Commit) error) error {
```
Comment (alcortesm) [inline commit_walker.go]: An awesome and very welcome simplification, thanks!

Using a queue here is quite dangerous though, as the garbage collector would not be able to free any of the already processed commits.  Consider using a stack instead, setting already processed commits' pointers to nil.

**Comment by mcuadros** [comment]
@smola what we do with this PR?

**Comment by mcuadros** [comment]
@smola can we retake this PR? I found interesting avoid having large stacks

**Comment by smola** [comment]
@mcuadros I'm playing around in a new branch (https://github.com/smola/go-git/commits/new-iters) to unify iterators implementations (or, at least, tree walkers), including TreeWalker, CommitWalker and the new difftree iterators.

**Comment by alcortesm** [comment]
+250 commits, lol.

**Pull request closed without merging**